src/Hook/Input/PrePush.php

```diff
@@ -55,13 +66,41 @@ private function createFromStdIn(array $stdIn): array
             [$localRef, $localHash, $remoteRef, $remoteHash] = explode(' ', trim($line));
 
             if (Util::isZeroHash($remoteHash)) {
-                continue;
+                $remoteHash = $this->getHashOfBranchOrigin();
+
+                if (Util::isZeroHash($remoteHash)) {
+                    continue;
+                }
             }
 
-            $from     = new Ref($remoteRef, $remoteHash, Util::extractBranchFromRefPath($remoteRef));
-            $to       = new Ref($localRef, $localHash, Util::extractBranchFromRefPath($localRef));
+            $from = new Ref($remoteRef, $remoteHash, Util::extractBranchFromRefPath($remoteRef));
+            $to = new Ref($localRef, $localHash, Util::extractBranchFromRefPath($localRef));
             $ranges[] = new Range($from, $to);
         }
+
         return $ranges;
     }
+
+    private function getHashOfBranchOrigin(): string
```

Comment:
This method is the actual fix. All other files are just changes because the repository is needed.
With `git reflog show` I get a list of all changes since the branch was created. 
This looks like this:
```
4cd8bca1a853cde29a8a238ea24a36cf64898661 (HEAD -> feature/test) feature/test@{0}: commit: Improvement 2
05b5141ca91790d2becac4465232fa8ad695793e feature/test@{1}: commit: Improvement 1
3e6bc3932b396aa86b8e7d67188944956701bc13 (origin/master, master) feature/test@{2}: branch: Created from HEAD
```

The bottom entry of this list is the creation of the branch and it's hash can be used to get a list of changed files since the branch was created.